src/contextmenu.js

```diff
@@ -0,0 +1,472 @@
+      const selectedBlocksLength = blockSelection.size;
+      if (scope.block.isCollapsed()) {
+        if (selectedBlocksLength <= 1) {
+          return Blockly.Msg['EXPAND_BLOCK'];
+        } else {
+          return Blockly.Msg['EXPAND_BLOCK'] + ' (' +
+          selectedBlocksLength + ')';
+        }
+      } else {
+        if (selectedBlocksLength <= 1) {
+          return Blockly.Msg['COLLAPSE_BLOCK'];
+        } else {
+          return Blockly.Msg['COLLAPSE_BLOCK'] + ' (' +
+          selectedBlocksLength + ')';
+        }
+      }
+    },
+    preconditionFn: function(scope) {
+      const block = scope.block;
+      if (!block.isInFlyout && block.isMovable() &&
+        block.workspace.options.collapse) {
+        return 'enabled';
+      }
+      return 'hidden';
+    },
+    callback: function(scope) {
+      const state = !scope.block.isCollapsed();
+      const apply = function(block) {
+        if (block && collapseExpandOption.preconditionFn({
+          block: block,
+        }) === 'enabled' && !hasSelectedParent(block)) {
+          block.setCollapsed(state);
+        }
+      };
+      Blockly.Events.setGroup(true);
+      if (scope.block && blockSelection.size === 0) {
+        apply(scope.block);
+      }
+      blockSelection.forEach(function(id) {
+        const block = Blockly.getMainWorkspace().getBlockById(id);
+        if (block) {
+          apply(block);
+        }
+      });
+      Blockly.Events.setGroup(false);
+    },
+    scopeType: Blockly.ContextMenuRegistry.ScopeType.BLOCK,
+    id: 'blockCollapseExpand',
+    weight: 4,
+  };
+  Blockly.ContextMenuRegistry.registry.register(collapseExpandOption);
+};
+
+/**
+ * Modification for context menu 'Disable' to be available for multiple blocks.
+ */
+const registerDisable = function() {
+  const disableOption = {
+    displayText: function(scope) {
+      const selectedBlocksLength = blockSelection.size;
+      if (scope.block.isEnabled()) {
+        if (selectedBlocksLength <= 1) {
+          return Blockly.Msg['DISABLE_BLOCK'];
+        } else {
+          return Blockly.Msg['DISABLE_BLOCK'] + ' (' +
+          selectedBlocksLength + ')';
+        }
+      } else {
+        if (selectedBlocksLength <= 1) {
+          return Blockly.Msg['ENABLE_BLOCK'];
+        } else {
+          return Blockly.Msg['ENABLE_BLOCK'] + ' (' +
+          selectedBlocksLength + ')';
+        }
+      }
+    },
+    preconditionFn: function(scope) {
+      const block = scope.block;
+      if (!block.isInFlyout && block.workspace.options.disable &&
+        block.isEditable()) {
+        if (block.getInheritedDisabled()) {
+          return 'disabled';
+        }
+        return 'enabled';
+      }
+      return 'hidden';
+    },
+    callback: function(scope) {
+      const state = !scope.block.isEnabled();
+      const apply = function(block) {
+        if (block && disableOption.preconditionFn({
+          block: block,
+        }) === 'enabled' && !hasSelectedParent(block)) {
```

Comment:
Similar to the collapse case above, I think you do want to enable blocks that have selected parents. But I don't think you want to disable blocks that have selected parents.